src/mono/sample/wasi/native/Program.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+public unsafe class Test
+{
+    [UnmanagedCallersOnly(EntryPoint = "ManagedFunc")]
+    public static int MyExport(int number)
```

Comment:
Agreed with Pavel that it might be good for the entrypoint and managed name to match, unless we are explicitly testing that you can make them not match and it will still work